tests/gnrc_ndp2/main.c

```diff
@@ -0,0 +1,1039 @@
+    TEST_ASSERT_NOT_NULL((pkt = gnrc_ndp2_nbr_adv_build(&test_tgt, flags,
+                                                        NULL)));
+    TEST_ASSERT(gnrc_pktbuf_is_sane());
+    /* check packet meta-data */
+    TEST_ASSERT_NULL(pkt->next);
+    TEST_ASSERT_NOT_NULL(pkt->data);
+    TEST_ASSERT_EQUAL_INT(sizeof(ndp_nbr_adv_t), pkt->size);
+    TEST_ASSERT_EQUAL_INT(GNRC_NETTYPE_ICMPV6, pkt->type);
+    /* check packet content */
+    nbr_adv = pkt->data;
+    TEST_ASSERT_EQUAL_INT(ICMPV6_NBR_ADV, nbr_adv->type);
+    TEST_ASSERT_EQUAL_INT(0, nbr_adv->code);
+    TEST_ASSERT_EQUAL_INT(flags, nbr_adv->flags);
+    TEST_ASSERT_EQUAL_INT(0, nbr_adv->resv[0]);
+    TEST_ASSERT_EQUAL_INT(0, nbr_adv->resv[1]);
+    TEST_ASSERT_EQUAL_INT(0, nbr_adv->resv[2]);
+    TEST_ASSERT_MESSAGE(ipv6_addr_equal(&test_tgt, &nbr_adv->tgt),
+                        "nbr_adv->tgt != test_tgt");
+    gnrc_pktbuf_release(pkt);
+    TEST_ASSERT(gnrc_pktbuf_is_empty());
+}
+
+static void test_nbr_adv_build__success_without_flags(void)
+{
+    test_nbr_adv_build__success(0);
+}
+
+static void test_nbr_adv_build__success_with_flags(void)
+{
+    test_nbr_adv_build__success(NDP_NBR_ADV_FLAGS_S);
+}
+
+static void test_rtr_sol_build__pktbuf_full(void)
+{
+    fill_pktbuf();
+    TEST_ASSERT_NULL(gnrc_ndp2_rtr_sol_build(NULL));
+}
+
+static void test_rtr_sol_build__success(void)
+{
+    gnrc_pktsnip_t *pkt;
+    ndp_rtr_sol_t *rtr_sol;
+
+    TEST_ASSERT(gnrc_pktbuf_is_empty());
+    TEST_ASSERT_NOT_NULL((pkt = gnrc_ndp2_rtr_sol_build(NULL)));
+    TEST_ASSERT(gnrc_pktbuf_is_sane());
+    /* check packet meta-data */
+    TEST_ASSERT_NULL(pkt->next);
+    TEST_ASSERT_NOT_NULL(pkt->data);
+    TEST_ASSERT_EQUAL_INT(sizeof(ndp_rtr_sol_t), pkt->size);
+    TEST_ASSERT_EQUAL_INT(GNRC_NETTYPE_ICMPV6, pkt->type);
+    /* check packet content */
+    rtr_sol = pkt->data;
+    TEST_ASSERT_EQUAL_INT(ICMPV6_RTR_SOL, rtr_sol->type);
+    TEST_ASSERT_EQUAL_INT(0, rtr_sol->code);
+    TEST_ASSERT_EQUAL_INT(0, rtr_sol->resv.u32);
+    gnrc_pktbuf_release(pkt);
+    TEST_ASSERT(gnrc_pktbuf_is_empty());
+}
+
+static void test_rtr_adv_build__pktbuf_full(void)
+{
+    fill_pktbuf();
+    TEST_ASSERT_NULL(gnrc_ndp2_rtr_adv_build(TEST_CUR_HL, 0, TEST_LTIME,
+                                             TEST_REACH_TIME,
+                                             TEST_RETRANS_TIMER, NULL));
+}
+
+static void test_rtr_adv_build__success(uint8_t flags)
+{
+    gnrc_pktsnip_t *pkt;
+    ndp_rtr_adv_t *rtr_adv;
+
+    TEST_ASSERT(gnrc_pktbuf_is_empty());
+    TEST_ASSERT_NOT_NULL((pkt = gnrc_ndp2_rtr_adv_build(TEST_CUR_HL, flags,
+                                                        TEST_LTIME,
+                                                        TEST_REACH_TIME,
+                                                        TEST_RETRANS_TIMER,
+                                                        NULL)));
+    TEST_ASSERT(gnrc_pktbuf_is_sane());
+    /* check packet meta-data */
```

Comment:
please provide a macro or inline function for this meta-data check to reduce code redundancy